lib/external_service_new/zendesk_downloader.rb

```diff
@@ -0,0 +1,51 @@
+module ExternalServiceNew
+	class ZendeskDownloader
+		def initialize(dispatcher:, logger:nil)
+			@dispatcher = dispatcher
+			@logger			= logger || respond_to?(:logger) ? logger : nil
+		end
+
+		def each_list(&block)
+			return to_enum(:each_list) unless block
+			[[
+				dispatcher.site_url,
+				"#{dispatcher.site_url}'s customers"
+			]].each(&block)
+		end
+
+		def each_email(id: nil, &block)
+			ZendeskInternalDownloader.new(
+				dispatcher: dispatcher, 
+				logger: logger
+			).each_email(&block)
+		end
+
+		private
+
+		attr_reader :dispatcher, :logger
+	end
+
+	class ZendeskInternalDownloader
+		def initialize(dispatcher:, filters: nil, logger:nil)	
+			raise "dispatcher must not be nil, found #{dispatcher.inspect}" unless dispatcher
```

Comment:
I also see that but as it's like that in all the other downloader i didn't think much